orttraining/orttraining/lazy_tensor/accelerator.cpp

```diff
@@ -0,0 +1,250 @@
+    case aten::div:
+    //case aten::gt:
+    //case aten::eq:
+    case prim::Constant:
+      //case aten::threshold_backward:
+      std::cout << "[compiler.cc] Support " << *node;  //<< std::endl;
+      return true;
+    default:
+      std::cout << "[compiler.cc] Not support " << *node;  //<< std::endl;
+      return false;
+  }
+}
+
+void Accelerator::Run(torch::jit::Stack& stack) {
+  // Get the number of expected inputs to the graph we are compiling
+  const at::ArrayRef<torch::jit::Value*>& graph_inputs = subgraph_->inputs();
+  const auto num_inputs = graph_inputs.size();
+
+  // Pop these inputs from the stack.
+  at::ArrayRef<c10::IValue> inputs = torch::jit::last(stack, num_inputs);
+
+  std::cout << "JIT sub-graph: " << std::endl;
+  std::cout << *subgraph_ << std::endl;
+  // If we haven't compiled for the shape/device of these inputs before,
+  // do so now.
+  torch::jit::CompleteArgumentSpec spec{false, at::ArrayRef<c10::IValue>(inputs)};
+  if (cache_.find(spec) == cache_.end()) {
+    cache_.emplace(spec, Compile(spec, inputs));
+  }
+
+  // Run the compiled function!
+  auto outputs = cache_[spec].code(inputs);
+
+  torch::jit::drop(stack, num_inputs);
+
+  for (auto& output : outputs) {
+    auto var = torch::autograd::make_variable(output.toTensor());
+    stack.push_back(c10::IValue(var));
+  }
+}
+
+void Accelerator::CheckArgs(
+    const at::ArrayRef<c10::IValue>& inputs) {
+  // TODO: remove this check.
+  TORCH_CHECK(inputs.size(), "Need at least one input.");
+  for (const auto& input : inputs) {
+    TORCH_CHECK(input.isTensor(), "Compiler can only handle Tensor inputs.");
+  }
+}
+
+// Store input types in sub-graph so that
+// ONNX exporter can use them. Input types
+// are required when executing ONNX model
+// in ORT.
+// TODO: Allow ORT to accept models without
+// input types. Then, we can remove this function.
+void Accelerator::PropagateArgTypes(
+    const at::ArrayRef<c10::IValue>& inputs) {
+  TORCH_CHECK(subgraph_->inputs().size() == inputs.size(),
+              "Number of provided inputs must match captured sub-graph's schema.");
+  const auto num_inputs = subgraph_->inputs().size();
+  for (size_t i = 0; i < num_inputs; ++i) {
+    auto input_symbol = subgraph_->inputs()[i];
+    auto input_value = inputs[i];
+    input_symbol->setType(input_value.type());
+  }
+}
+
+// ONNX exporter is written in Python, so
+// this function may calls some Python functions.
+// Be aware of GIL issue.
+// The returned value is the path to exported
+// ONNX file.
+static std::string ExportToOnnx(std::shared_ptr<torch::jit::Graph> graph) {
+  pybind11::gil_scoped_acquire guard{};
+  // Retrieve Python exporter function.
+  pybind11::function export_to_onnx =
+      pybind11::reinterpret_borrow<pybind11::function>(
+          pybind11::module::import("torch.onnx.utils").attr("_optimize_graph_1"));
+  // Execute Python function.
+  auto result = export_to_onnx(graph, ::torch::onnx::OperatorExportTypes::ONNX);
```

Comment:
Where the onnx file is created? Is it removed?

Reply:
It's removed now.